src/foundry/common/abstract/document.mjs.d.ts

```diff
@@ -370,7 +371,7 @@ declare abstract class Document<
    * @param embeddedName - The name of the embedded Document type
    * @returns The Collection instance of embedded Documents of the requested type
    */
-  getEmbeddedCollection(embeddedName: string): Collection<Document<any, this>>; // TODO: Improve
+  getEmbeddedCollection(embeddedName: string): EmbeddedCollection<DocumentConstructor, AnyDocumentData>; // TODO: Improve
```

Comment:
I needed to change this to type `Token.getEmbeddedCollection` correctly.
Note: `EmbeddedCollection<DocumentConstructor, ConcreteDocumentData>` was not possible because the `TokenDocument` returns collections of embeddable that parents are Actors.

Reply:
yup, makes sense :+1: